lib/modules/manager/gradle/artifacts.ts

```diff
@@ -153,17 +187,23 @@ export async function updateArtifacts({
       !updatedDeps.length ||
       isGcvPropsFile(packageFileName)
     ) {
-      cmd += ' --write-locks';
-    } else {
+      lockfileCmd += ' --write-locks';
+    } else if (lockFiles.length) {
       const updatedDepNames = updatedDeps
         .map(({ depName, packageName }) => packageName ?? depName)
         .filter(is.nonEmptyStringAndNotWhitespace);
 
-      cmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
+      lockfileCmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
     }
 
     await writeLocalFile(packageFileName, newPackageFileContent);
-    await exec(cmd, { ...execOptions, ignoreStdout: true });
+    await exec(lockfileCmd, { ...execOptions, ignoreStdout: true });
+
+    await updateVerificationMetadata(
+      verificationMetadataFile,
+      baseCmd,
+      execOptions
+    );
```

Comment:
You should combine both commands to a single exec call, you can pass an array of commands.
This will improve performance when on docker mode (eg: hosted app).
Currently this will cause two docker sidecar container, which both need to install tools.